rust/core-lib/src/editor.rs

```diff
@@ -913,27 +930,35 @@ impl<W: Write + Send + 'static> Editor<W> {
         self.text.measure::<LinesMetric>() + 1
     }
 
-    pub fn plugin_set_fg_spans(&mut self, start: usize, len: usize, spans: Vec<Span>, rev: usize) {
+    pub fn plugin_add_scopes(&mut self, plugin: &str, scopes: Vec<Vec<String>>) {
+        self.style_scopes.add_scopes(plugin, scopes, &self.doc_ctx);
```

Comment:
In the future it would be nice if each plugin could have more than one layer, but in practice one layer per plugin will cover 99% of use cases and everything except large multi-functional plugins that maybe should be split into separate plugins. And the only functionality they lose is being able to separately dispose of and add spans for different things from the same plugin. So this is totally fine for now, and as a default.